src/fix/main.cpp

```diff
@@ -1257,42 +1258,23 @@ int main(int argc, char *argv[]) {
 			fixSpec = 0;
 			while (*musl_optarg) {
 				switch (*musl_optarg) {
-#define SPEC_l FIX_LOGO
-#define SPEC_L TRASH_LOGO
-#define SPEC_h FIX_HEADER_SUM
-#define SPEC_H TRASH_HEADER_SUM
-#define SPEC_g FIX_GLOBAL_SUM
-#define SPEC_G TRASH_GLOBAL_SUM
-#define overrideSpec(cur, bad) \
-	do { \
-		if (fixSpec & CAT(SPEC_, bad)) \
+#define OVERRIDE_SPEC(cur, bad, curFlag, badFlag) \
+	case STR(cur)[0]: \
+		if (fixSpec & badFlag) \
 			fprintf(stderr, "warning: '" STR(cur) "' overriding '" STR(bad) "' in fix spec\n"); \
-		fixSpec = (fixSpec & ~CAT(SPEC_, bad)) | CAT(SPEC_, cur); \
-	} while (0)
-				case 'l':
-					overrideSpec(l, L);
-					break;
-				case 'L':
-					overrideSpec(L, l);
-					break;
-
-				case 'h':
-					overrideSpec(h, H);
-					break;
-				case 'H':
-					overrideSpec(H, h);
-					break;
-
-				case 'g':
-					overrideSpec(g, G);
-					break;
-				case 'G':
-					overrideSpec(G, g);
-					break;
+		fixSpec = (fixSpec & ~badFlag) | curFlag; \
+		break
+#define overrideSpecs(fix, fixFlag, trash, trashFlag) \
+	OVERRIDE_SPEC(fix, trash, fixFlag, trashFlag); \
+	OVERRIDE_SPEC(trash, fix, trashFlag, fixFlag)
+				overrideSpecs(l, FIX_LOGO,       L, TRASH_LOGO);
+				overrideSpecs(h, FIX_HEADER_SUM, H, TRASH_HEADER_SUM);
+				overrideSpecs(g, FIX_GLOBAL_SUM, G, TRASH_GLOBAL_SUM);
```

Comment:
I considered doing like `overrideSpecs(l, L, LOGO);` and then internally doing `CAT(FIX_, name)` and `CAT(TRASH_, name)`, but that would make it harder to grep for the exact constant names that were originally `#define`d.